File: server/router/system.go
package router

import (
	"github.com/lihongsheng/go-admin/server/api/v1/base"
	"github.com/lihongsheng/go-admin/server/api/v1/system"
	"github.com/lihongsheng/go-admin/server/middleware"

	"github.com/gin-gonic/gin"
)

// BaseRouter 登录 / 验证码 / 当前用户 / 当前菜单
func BaseRouter(g *gin.RouterGroup) {
	b := g.Group("/base")
	b.GET("/captcha", base.Captcha)
	b.POST("/login", base.Login)
	auth := b.Group("", middleware.JWTAuth())
	auth.POST("/logout", base.Logout)
	auth.GET("/info", base.Info)
	auth.GET("/menu", base.Menu)

	// 上传接口：登录用户均可上传，不走 CasbinAuth
	upload := b.Group("/upload", middleware.JWTAuth())
	upload.POST("", base.Upload)
	baseSystem := b.Group("/system", middleware.JWTAuth())
	baseSystem.GET("/types", base.GetSystemTypeInfo)
}

// SystemRouter 用户 / 角色 / 菜单 / API
func SystemRouter(g *gin.RouterGroup) {
	s := g.Group("/system", middleware.JWTAuth(), middleware.CasbinAuth())

	s.POST("/user", system.UserCreate)
	s.PUT("/user", system.UserUpdate)
	s.DELETE("/user/:id", system.UserDelete)
	s.GET("/user/list", system.UserList)

	s.POST("/role", system.RoleCreate)
	s.PUT("/role", system.RoleUpdate)
	s.DELETE("/role/:id", system.RoleDelete)
	s.GET("/role/list", system.RoleList)
	s.POST("/role/auth", system.RoleAuth)
	s.GET("/role/auth/:id", system.RoleAuthDetail)
	s.PUT("/role/:id/default-router", system.RoleSetDefaultRouter)

	s.POST("/menu", system.MenuCreate)
	s.PUT("/menu", system.MenuUpdate)
	s.DELETE("/menu/:id", system.MenuDelete)
	s.GET("/menu/tree", system.MenuTree)

	s.POST("/mch", system.MchCreate)
	s.PUT("/mch", system.MchUpdate)
	s.GET("/mch/:id", system.MchDetail)
	s.GET("/mch/no/:mchNo", system.MchDetailByNo)
	s.GET("/mch/list", system.MchList)
}

File: web/src/api/system.js
import request from '@/utils/request'

// user
export const userList   = (params) => request.get('/api/v1/system/user/list', { params })
export const userCreate = (data)   => request.post('/api/v1/system/user', data)
export const userUpdate = (data)   => request.put('/api/v1/system/user', data)
export const userDelete = (id)     => request.delete('/api/v1/system/user/' + id)

// upload
export const uploadFile = (file) => {
  const fd = new FormData()
  fd.append('file', file)
  return request.post('/api/v1/base/upload', fd, {
    headers: { 'Content-Type': 'multipart/form-data' }
  })
}

// role
export const roleList       = ()        => request.get('/api/v1/system/role/list')
export const roleCreate     = (data)    => request.post('/api/v1/system/role', data)
export const roleUpdate     = (data)    => request.put('/api/v1/system/role', data)
export const roleDelete     = (id)      => request.delete('/api/v1/system/role/' + id)
export const roleAuth            = (data)    => request.post('/api/v1/system/role/auth', data)
export const roleAuthDetail      = (id)      => request.get('/api/v1/system/role/auth/' + id)
export const roleSetDefaultRouter = (id, data) => request.put('/api/v1/system/role/' + id + '/default-router', data)

// menu
export const menuTree   = ()        => request.get('/api/v1/system/menu/tree')
export const menuCreate = (data)    => request.post('/api/v1/system/menu', data)
export const menuUpdate = (data)    => request.put('/api/v1/system/menu', data)
export const menuDelete = (id)      => request.delete('/api/v1/system/menu/' + id)

// api
export const apiList    = (params)  => request.get('/api/v1/system/api/list', { params })
export const apiCreate  = (data)    => request.post('/api/v1/system/api', data)
export const apiUpdate  = (data)    => request.put('/api/v1/system/api', data)
export const apiDelete  = (id)      => request.delete('/api/v1/system/api/' + id)

// plugin
export const pluginList = ()        => request.get('/api/v1/plugin/list')
